mempool/src/shared_mempool/tasks.rs

```diff
@@ -236,8 +236,28 @@ pub(crate) async fn process_transaction_broadcast<V>(
 ) where
     V: TransactionValidation,
 {
-    let results = process_incoming_transactions(&smp, transactions, timeline_state).await;
+    // process transactions and log the result
+    let results = process_incoming_transactions(&smp, transactions.clone(), timeline_state).await;
     log_txn_process_results(&results, Some(peer.peer_id()));
+
+    // we assume that the results contains a vec of the same length of transactions
+    debug_assert_eq!(results.len(), transactions.len());
+
+    // security log all the failed transactions
+    let failed_transactions = results
+        .iter()
+        // create (MempoolStatus, transaction) iterator
```

Comment:
nit: This comment confused me initially. I'd maybe update this comment to either: (i) change MempoolStatus to SubmissionStatus (as that is the type returned by process_incoming_transactions); or (ii) I'd be more explicit and just say:
// create ((MempoolStatus, Option<VMStatus>), transaction) iterator. It